File: news_server/modules/source/infrastructure/getters/juejin.ts
/**
 * 稀土掘金新闻源获取器
 * 从 newsnow/server/sources/juejin.ts 迁移
 */
import type { SourceGetter } from '../../domain/getter/getter'
import { NewsItem } from '../../domain/newsitem/entity'

interface JuejinResponse {
  data: {
    content: {
      title: string
      content_id: string
    }
  }[]
}

export const juejinGetter: SourceGetter = async () => {
  const url = 'https://api.juejin.cn/content_api/v1/content/article_rank?category_id=1&type=hot&spider=0'
  const response = await fetch(url)
  
  if (!response.ok) {
    throw new Error(`Failed to fetch Juejin: ${response.statusText}`)
  }

  const res = (await response.json()) as JuejinResponse
  
  return res.data.map((k) =>
    NewsItem.fromEntity({
      id: k.content.content_id,
      title: k.content.title,
      url: `https://juejin.cn/post/${k.content.content_id}`,
    })
  )
}

File: modules/source/infrastructure/getters/juejin.go
package getters

import "context"

func (r *Registry) juejin(ctx context.Context) ([]Item, error) {
	var res struct {
		Data []struct {
			Content struct {
				Title     string `json:"title"`
				ContentID string `json:"content_id"`
			} `json:"content"`
		} `json:"data"`
	}
	if err := r.getJSON(ctx, "https://api.juejin.cn/content_api/v1/content/article_rank?category_id=1&type=hot&spider=0", nil, &res); err != nil {
		return nil, err
	}
	out := make([]Item, 0, len(res.Data))
	for _, k := range res.Data {
		id := k.Content.ContentID
		if id == "" {
			continue
		}
		out = append(out, Item{ID: id, Title: k.Content.Title, URL: "https://juejin.cn/post/" + id})
	}
	return out, nil
}
